Source/ExecutionEngine/ExecutionEngine.cs

```diff
@@ -804,23 +804,11 @@ private Implementation[] GetPrioritizedImplementations(Program program)
         foreach (var task in tasks) {
           task.Result.ProcessXml(this);
         }
-      }
-      catch (AggregateException ae) {
-        ae.Flatten().Handle(e =>
-        {
-          if (e is ProverException) {
-            Options.Printer.ErrorWriteLine(Console.Out, "Fatal Error: ProverException: {0}", e.Message);
-            outcome = PipelineOutcome.FatalError;
-            return true;
-          }
-
-          if (e is OperationCanceledException) {
-            outcome = PipelineOutcome.Cancelled;
-            return true;
-          }
-
-          return false;
-        });
+      } catch(TaskCanceledException) {
+        outcome = PipelineOutcome.Cancelled;
+      } catch(ProverException e) {
+        Options.Printer.ErrorWriteLine(Console.Out, "Fatal Error: ProverException: {0}", e.Message);
+        outcome = PipelineOutcome.FatalError;
```

Comment:
Do these exceptions no longer occur in `AggregateException` bundles?

Reply:
I think so. An AggregateException can be created when synchronously blocking on a Task, like when using `.Result`, of which two have been removed in this PR. Without this change a test failed.